### PR TITLE
Update regex of email validation

### PR DESCRIPTION
**Pull request by AlaaElattar**
- https://github.com/threefoldtech/threefold_connect/issues/541
- Update verification to ensure that there’s a dot followed by at least two letters at the end, which prevents invalid entries like `aa@aa`.

### Tested Scenarios:
All the following cases will be rejected by regex:
- `aa@aa`
- `user..name@gmail.com`
- `username@domain-.com`
- `username@-domain.com`
- `.user@gmail.com`
- `user.@gmail.com`

### CODE REVIEW

**Review by zaelgohary — CHANGES_REQUESTED** [review]
The regex accepts the following invalid scenarios:

- Two or more consecutive dots in the local part

image

- Invalid max length

image

- Hyphens at invalid positions in the domain

image

image

- Leading or trailing dot in the local part

image

image

**Comment by AlaaElattar** [comment]
image
image
image
image
image

**Review by zaelgohary — COMMENTED** [review]
Can we use email validator instead of regex?

File: app/lib/services/tools_service.dart [hunk 1]
```diff
@@ -19,11 +20,14 @@ String randomString(int len) {
...
 
 bool validateEmail(String? value) {
-  RegExp regex = RegExp(
-      r"^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,253}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,253}[a-zA-Z0-9])?)*$");
-  return regex.hasMatch(value.toString());
+  if (value == null || value.isEmpty) {
+    return false;
+  }
+
+  return EmailValidator.validate(value);
```
Comment (zaelgohary) [inline app/lib/services/tools_service.dart:27]: Can we use isEmail from validators instead?

**Review by zaelgohary — APPROVED** [review]

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**